packages/forms/test/form_array_spec.ts

```diff
@@ -64,6 +66,98 @@ describe('FormArray', () => {
 
       expect(a.controls).toEqual([c1, c2, c3]);
     });
+
+    it('should not emit events when calling `FormArray.push` with `emitEvent: false`', () => {
+      a.valueChanges.subscribe(() => logger.push('value change'));
+      a.statusChanges.subscribe(() => logger.push('status change'));
+
+      a.push(c1, {emitEvent: false});
+
+      expect(a.length).toEqual(1);
+      expect(a.controls).toEqual([c1]);
+      expect(logger).toEqual([]);
+    });
+
+    it('should not emit events when calling `FormArray.removeAt` with `emitEvent: false`', () => {
+      a.push(c1);
+      a.push(c2);
+      a.push(c3);
+
+      a.valueChanges.subscribe(() => logger.push('value change'));
+      a.statusChanges.subscribe(() => logger.push('status change'));
+
+      a.removeAt(1, {emitEvent: false});
+
+      expect(a.controls).toEqual([c1, c3]);
+      expect(logger).toEqual([]);
+    });
+
+    it('should not emit events when calling `FormArray.clear` with `emitEvent: false`', () => {
+      a.push(c1);
+      a.push(c2);
+      a.push(c3);
+
+      a.valueChanges.subscribe(() => logger.push('value change'));
+      a.statusChanges.subscribe(() => logger.push('status change'));
+
+      a.clear({emitEvent: false});
+
+      expect(a.controls).toEqual([]);
+      expect(logger).toEqual([]);
+    });
+
+    it('should not emit events when calling `FormArray.insert` with `emitEvent: false`', () => {
+      a.push(c1);
+      a.push(c3);
+
+      a.valueChanges.subscribe(() => logger.push('value change'));
+      a.statusChanges.subscribe(() => logger.push('status change'));
+
+      a.insert(1, c2, {emitEvent: false});
+
+      expect(a.controls).toEqual([c1, c2, c3]);
+      expect(logger).toEqual([]);
+    });
+
+    it('should not emit events when calling `FormArray.setControl` with `emitEvent: false`', () => {
+      a.push(c1);
+      a.push(c3);
+
+      a.valueChanges.subscribe(() => logger.push('value change'));
+      a.statusChanges.subscribe(() => logger.push('status change'));
+
+      a.setControl(1, c2, {emitEvent: false});
+
+      expect(a.controls).toEqual([c1, c2]);
+      expect(logger).toEqual([]);
+    });
+
+    it('should not emit status change events when `FormArray.push` is called with `emitEvent: false`',
+       () => {
+         const validatorFn = (value: any) => value.controls.length > 0 ? {controls: true} : null;
```

Comment:
Could you plz add a comment on why we've added validators here?
```suggestion
         // Adding validators to make sure there are no status change event submitted when form becomes invalid.
         const validatorFn = (value: any) => value.controls.length > 0 ? {controls: true} : null;
```